pkg/controller/storagecluster/controller_test.go

```diff
@@ -6238,6 +6238,236 @@ func TestNodeShouldRunStoragePod(t *testing.T) {
+			match: true,
+		},
+		//{
+		//	old: &corev1.StorageClusterSpec{},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{},
+		//	},
+		//	match: true,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{},
+		//	},
+		//	new:   &corev1.StorageClusterSpec{},
+		//	match: true,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{},
+		//	},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: false,
+		//			}},
+		//	},
+		//	match: true,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: false,
+		//			}},
+		//	},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{},
+		//	},
+		//	match: true,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//			}},
+		//	},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{},
+		//	},
+		//	match: false,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{},
+		//	},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//			}},
+		//	},
+		//	match: false,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//			}},
+		//	},
+		//	match: false,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: false,
+		//			}},
+		//	},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//			}},
+		//	},
+		//	match: false,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//			}},
+		//	},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//			}},
+		//	},
+		//	match: true,
+		//},
+		//{
+		//	old: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//				Image:   "foo",
+		//			}},
+		//	},
+		//	new: &corev1.StorageClusterSpec{
+		//		Monitoring: &corev1.MonitoringSpec{
+		//			Telemetry: &corev1.TelemetrySpec{
+		//				Enabled: true,
+		//				Image:   "bar",
+		//			}},
+		//	},
+		//	match: false,
+		//},
+	}
```

Comment:
Was this left commented accidentally?
Can you update this as well as the cherry-pick>